src/ui/hook/useTimeout.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+type ClearTimeout = () => void
+type ResetTimeout = () => void
+type UseTimeout = [ClearTimeout, ResetTimeout]
+
+export const useTimeout = (callback: () => void, delay: number | null): UseTimeout => {
+  const timeoutRef = useRef<number | null>(null)
+
+  const set = useCallback(() => {
+    timeoutRef.current = delay && setTimeout(() => callback(), delay)
+  }, [callback, delay])
+
+  const clear = useCallback(() => {
+    timeoutRef.current && clearTimeout(timeoutRef.current)
```

Comment:
it would be a good practice to also set timeoutRef.current to null after clearing the timeout. This is to ensure the reference does not hold onto the timeout id once it has been cleared.

Reply:
I'm not sure it's needed in this case seeing as it's part of a hook.

Reply:
It's not necessary, but it's a good practice :)

Reply:
It's not necessary, but it's a good practice :)